### PR TITLE
update pgsql instrumenting to 8.1 by using objects instead of resources

### PR DESCRIPTION
**Pull request by ZNeumann**

### CODE REVIEW

File: agent/php_internal_instrument.c [hunk 1]
```diff
@@ -1574,13 +1582,20 @@ static char* nr_php_prepared_statement_make_pgsql_key(
...
 
+#if ZEND_MODULE_API_NO >= ZEND_8_1_X_API_NO /* PHP 8.1+ */
   if (FAILURE
       != zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET,
-                                  ZEND_NUM_ARGS() TSRMLS_CC, "|r", &pgsql)) {
+                                  ZEND_NUM_ARGS() TSRMLS_CC, "|o", &pgsql)) {
+#else
+  if (FAILURE
+      != zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET,
+                                  ZEND_NUM_ARGS() TSRMLS_CC, "|o", &pgsql)) {
```
Comment (mfulb) [inline agent/php_internal_instrument.c:1597]: Should this be "|r" like the original code since this is the != PHP8.1 case?

Comment (ZNeumann) [reply agent/php_internal_instrument.c]: Fixed

**Comment by zsistla** [comment]
ok jenkins

**Review by zsistla — APPROVED** [review]
Looks good, thanks for finding/fixing!

**Pull request merged**